security-command-center/snippets/src/test/java/management/api/SecurityHealthAnalyticsCustomModuleTest.java

```diff
@@ -0,0 +1,237 @@
+  private static ByteArrayOutputStream stdOut;
+
+  @Rule
+  public final MultipleAttemptsRule multipleAttemptsRule =
+      new MultipleAttemptsRule(MAX_ATTEMPT_COUNT, INITIAL_BACKOFF_MILLIS);
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp() throws IOException, InterruptedException {
+    final PrintStream out = System.out;
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("SCC_PROJECT_ORG_ID");
+
+    // Perform cleanup before running tests
+    cleanupExistingCustomModules();
+
+    stdOut = null;
+    System.setOut(out);
+    TimeUnit.MINUTES.sleep(3);
+  }
+
+  @AfterClass
+  public static void cleanUp() {
+    final PrintStream out = System.out;
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+    stdOut = null;
+    System.setOut(out);
+  }
+
+  @Before
+  public void beforeEach() {
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+  }
+
+  @After
+  public void afterEach() {
+    stdOut = null;
+    System.setOut(null);
+  }
+
+  // cleanupExistingCustomModules clean up all the existing custom module
+  public static void cleanupExistingCustomModules() throws IOException {
+    String parent = String.format("organizations/%s/locations/%s", ORGANIZATION_ID, LOCATION);
+
+    try (SecurityCenterManagementClient client = SecurityCenterManagementClient.create()) {
+
+      // create the request
+      ListSecurityHealthAnalyticsCustomModulesRequest request =
+          ListSecurityHealthAnalyticsCustomModulesRequest.newBuilder().setParent(parent).build();
+
+      // calls the API
+      ListSecurityHealthAnalyticsCustomModulesPagedResponse response =
+          client.listSecurityHealthAnalyticsCustomModules(request);
+
+      // Iterate over the response and delete custom module one by one which start with
+      // java_sample_custom_module
+      for (SecurityHealthAnalyticsCustomModule module : response.iterateAll()) {
+        if (module.getDisplayName().startsWith("java_sample_custom_module")) {
+          String customModuleId = extractCustomModuleId(module.getName());
+          deleteCustomModule(parent, customModuleId);
+        }
+      }
+    }
+  }
+
+  // extractCustomModuleID extracts the custom module Id from the full name
+  public static String extractCustomModuleId(String customModuleFullName) {
+    if (!Strings.isNullOrEmpty(customModuleFullName)) {
+      String[] result = customModuleFullName.split("/");
+      if (result.length > 0) {
+        return result[result.length - 1];
+      }
+    }
+    return "";
```

Comment:
nit: recommend to use RegEx instead to match and retrieve the module ID

Reply:
Addressed